RearrangeCarGroup8Movements.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Jul 22 10:37:40 2017
+Rearrange the car assignment
+Group 8 solution!
+"""
+
+import logging
+import random
+
+'''Given: There is a parking lot with N spaces and N-1 cars in it.
+The parking lot is described by an array of numbers, that is a permutation of numbers
+from 0 to N. Cars are with numbers from 1 to N-1, 0 means an empty parking space.
+
+E.G. Initial order = [1, 2, 0, 3]
+       Final order = [3, 1, 2, 0]
+
+Task: to rearrange the cars in a given way
+and print a series of moves from position x to y
+
+For our example, the series of moves should be e.g.:
+    move 1 to 2
+    move 3 to 0
+'''
+
+
+class Movement(object):
+    """This class stores information about one movement,
+    i.e. a car moving from position *from_index* to position *to_index*.
+    """
+
+    def __init__(self, car, from_index, to_index):
+        self.car = car
+        self.from_index = from_index
+        self.to_index = to_index
+
+    def __str__(self):
+        move_str = "Move car #{0} from {1} to {2}".format(self.car, self.from_index, self.to_index)
+        return move_str
+
+
+class MovementPrinter(object):
+    """
+    Class used to print the list of movements into stdout or log file.
+    """
+
+    def __init__(self, init_order, final_order, print_console):
+        self.order_message = self._prepare_order_message(init_order, final_order)
+        self.print_console = print_console
+
+    def _prepare_order_message(self, initial_car_order, final_car_order):
+        return ('Initial car order: {init_order}'.format(init_order=initial_car_order) +
+                'Final car order: {final_order}'.format(final_order=final_car_order) +
+                'Moves:')
+
+    def print_moves(self, moves):

```

Comment:
I think that in the code we should also deal with the case that the initial_car_order is already the same of the final_car_order and hence no moves have to be done.